### PR TITLE
`db/admin migrate`: Migrate first migration explicitly if DB version's migration is squashed

### PR DESCRIPTION
**Pull request by zrhoffman**
This PR fixes an issue encountered migrating the database if a migration for the current DB migration version does not exist, typically because a migration at that version has been squashed into `create_table.sql`:

```
trafficops-db-admin_1  | error: no migration found for version 2021012200000000: read down for version 2021012200000000 : file does not existError running migrate up: no migration found for version 2021012200000000: read down for version 2021012200000000 : file does not exist
trafficops-db-admin_1  | DB upgrade failed!
```

-----------------------------------------------

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops

## What is the best way to verify this PR?

1. Migrate the DB using Goose at [`d96c13c~`](https://github.com/apache/trafficcontrol/commit/d96c13cb40~) 
2. Dump the database and move the dump to `traffic_ops_db/test/docker/initdb.d`
3. Run the Traffic Ops DB tests

## If this is a bugfix, which Traffic Control versions contained the bug?

- master branch only

## PR submission checklist
- [x] Existing tests are sufficient 
- [x] This PR has documentation 
- [ ] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_ops/app/db/migrations/SQUASH.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+* `3_another-migration.up.sql`
+* `3_another-migration.down.sql`
+* `4_migration-name.up.sql`
+* `4_migration-name.down.sql`
+* `9_add-column-to-table.up.sql`
+* `9_add-column-to-table.down.sql`
+
+1. In order to prepare database migrations for the next major release, in this case, ATC 148.0.0, migrations `1` and `3` should be collapsed into `create_tables.sql` and migrations `4` and `9` should remain in [`traffic_ops/app/db/migrations/`](https://github.com/apache/trafficcontrol/tree/master/traffic_ops/app/db/migrations/).
+
+2. Note that `3` is the migration version version of the last up/down migration set. Find the definition for `LastSquashedMigrationVersion` in [`traffic_ops/app/db/admin.go`](https://github.com/apache/trafficcontrol/blob/master/traffic_ops/app/db/admin.go) and change it to `3`.
```
Comment (rimashah25) [inline traffic_ops/app/db/migrations/SQUASH.md:23]: version repeated twice.

File: traffic_ops/app/db/migrations/SQUASH.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Past PRs that have collapsed the DB migrations:
+- https://github.com/apache/trafficcontrol/pull/6065
+- https://github.com/apache/trafficcontrol/pull/3524
+
+<!--
```
Comment (rimashah25) [inline traffic_ops/app/db/migrations/SQUASH.md:29]: Shouldn't license information be on the top, to be in-sync with other files?

File: traffic_ops/app/db/migrations/SQUASH.md (hunk 1 above)
Comment (rimashah25) [reply traffic_ops/app/db/migrations/SQUASH.md:23]: Is there a way to automate this manual step? For eg: I haven't read create_tables.sql,  but when create_tables.sql runs and squashes/collapses previous migration, last migration date is stored. To remember to always update this date, I feel a user may easily forget.

Comment (zrhoffman) [reply traffic_ops/app/db/migrations/SQUASH.md]: > version repeated twice.

Removed one *version* in 84f2871c99

File: traffic_ops/app/db/migrations/SQUASH.md (hunk 2 above)
Comment (zrhoffman) [reply traffic_ops/app/db/migrations/SQUASH.md]: Moved the Apache License 2.0 to he top in c2376e06fe

File: traffic_ops/app/db/migrations/SQUASH.md (hunk 1 above)
Comment (zrhoffman) [reply traffic_ops/app/db/migrations/SQUASH.md]: > Is there a way to automate this manual step? For eg: I haven't read create_tables.sql, but when create_tables.sql runs and squashes/collapses previous migration, last migration date is stored. To remember to always update this date, I feel a user may easily forget.

Added `squash_migrations.sh`, which automates squashing migrations from the last ATC release, in 3de5dfdb08. We will lose `create_tables.sql` idempotency, which is really only an issue when two Traffic Ops instances are running the Postinstall script simultaneously, which is what #4984 was about. IMO this is not a big issue, but I filed #6106 so we don't lose track of it.

**Comment by rimashah25** [comment]
LGTM. Ready for merge.

**Review by rimashah25 — APPROVED** [review]

**Pull request merged**